cogito/protocol.go

```diff
@@ -212,18 +216,43 @@ func (src *Source) UnmarshalJSON(data []byte) error {
 // Validate verifies the Source configuration and applies defaults.
 func (src *Source) Validate() error {
 	//
-	// Validate mandatory fields.
+	// Evaluate mandatory fields.
 	//
 	var mandatory []string
-	if src.Owner == "" {
-		mandatory = append(mandatory, "owner")
+
+	defaultSinks := []string{"gchat", "github"}
+	defaultSinksSet := sets.From(defaultSinks...)
+	sinksSet := sets.From(src.Sinks...)
+
+	if sinksSet.Size() > 0 {
+		// First validate sinks are known and supported.
+		sinksNotValid := sinksSet.Difference(defaultSinksSet)
+		if sinksNotValid.Size() > 0 {
+			return fmt.Errorf("source: invalid sink: %s. Supported sinks: %s", sinksNotValid, defaultSinks)
+		}
 	}
-	if src.Repo == "" {
-		mandatory = append(mandatory, "repo")
+
+	if sinksSet.Size() == 0 || sinksSet.Contains("github") {
```

Comment:
Ufff I was just starting to review the previous code with `reflect`. This is much clearer to review :)

Reply:
Also, I would put this comment `// No sinks implies backward compatibility mode where github is mandatory and gchat optional.` above if statement not below.